### PR TITLE
feat(levm): implement blobhash logic

### LINKED ISSUE #933: Levm: Implement BLOBHASH logic
Validations are made in op_blobhash function, but the funcionality is not implemented, and would panic if the program reaches that point:
``` Rust
        // Should push in stack the blob hash
        unimplemented!("when we have tx implemented");

```

### PR DESCRIPTION
**Pull request by maximopalopoli**
**Motivation**

The motivation is to implement the blobhash opcode, since was not implemented yet.

**Description**

Main changes:
- Adds an atribute to store the blob hashes in Environment type
- implements blobhash logic, basicly taking an element of the vec of blob hashes and pushing it to the stack
- To do that, add some posible errors to VMError enum

Closes #933

### CODE REVIEW

File: crates/vm/levm/src/opcode_handlers/block.rs [hunk 1]
```diff
@@ -145,16 +146,35 @@ impl VM {
...
+            Some(vec) => match vec.get(index) {
+                Some(el) => *el,
+                None => {
+                    return Err(VMError::TransactionDoesNotHaveABlobHashVector);
+                }
+            },
+            None => {
+                return Err(VMError::NotEnoughBlobHashes);
+            }
+        };
```
Comment (JereSalo) [inline crates/vm/levm/src/opcode_handlers/block.rs:170]: I think that the errors are inverted here. The first match should return the error of "**missing blob hash vector**" and the internal match should return the "**blob hash not found**" error.
I would also rename these errors, some suggestions are:
**MissingBlobHashes** - when the vector is not found
**BlobHashIndexOutOfBounds** - when the blobhash is not found

**Review by JereSalo — CHANGES_REQUESTED** [review]
It looks good overall. I made a few suggestions.

File: crates/vm/levm/src/opcode_handlers/block.rs [hunk 2]
```diff
@@ -145,16 +146,35 @@ impl VM {
...
+    /// Currently not tested
     pub fn op_blobhash(
         &mut self,
         current_call_frame: &mut CallFrame,
     ) -> Result<OpcodeSuccess, VMError> {
         self.increase_consumed_gas(current_call_frame, gas_cost::BLOBHASH)?;
 
-        // Should push in stack the blob hash
-        unimplemented!("when we have tx implemented");
+        let index = current_call_frame.stack.pop()?;
```
[the remark this answers is not in the document]
Comment (maximopalopoli) [reply crates/vm/levm/src/opcode_handlers/block.rs:156]: Yes, i thinked maybe was unaesthetic to use a . after a ?, but if you suggest it, it shouldn't be.

File: crates/vm/levm/src/opcode_handlers/block.rs (hunk 1 above)
Comment (maximopalopoli) [reply crates/vm/levm/src/opcode_handlers/block.rs]: Done!

**Review by JereSalo — APPROVED** [review]
Looks good to me! Just fix the clippy lint and that's it.

**Review by jrchatruc — APPROVED** [review]

**Pull request merged**